### PR TITLE
ATB 1125 + ATB 1126 | Establish pipelines in dev, build and staging for alarms and create GitHub workflows for Alarm template

### PR DESCRIPTION
**Pull request by MinaMahdi**
## Proposed changes

ATB -1125: Establish Alarm template workflow
ATB-1126: Establish core, main, and alarm pipeline in staging along with alarm pipeline in dev and build

### What changed

Established secure pipelines in build, dev, and staging environments as follows: 
Dev: ais-alarm-pipeline 
Build: ais-alarm-pipeline
Staging: ais-core-pipeline , ais-main-pipeline , ais-alarm-pipeline

Created Github workflows for our Alarm template - manual-publish-alarm-to-dev.yaml + merge-alarm-to-main.yaml

### Why did it change

To provide pipelines for the Alarm stack in dev, build and staging and a workflow for our Alarm template. 

### Issue tracking

- [ATB-1126](https://govukverify.atlassian.net/browse/ATB-1126)

## Checklists
- [ ] Did not commit any not-required changes to the `src/infra/**/samconfig.toml`
- [ ] Tested changes and included test evidence in the PR, if appropriate
- [ ] Included all required tags and other properties for any new resources in the SAM template
- [ ] Ensured that any new resources in the SAM Template follow appropriate naming conventions
- [ ] Ensured that naming of new resources is compatible with deploying multiple stacks with custom stack names during development
- [ ] Ensured that no log lines include PII or other sensitive data
- [ ] Implemented unit testing for any new logic implemented, if appropriate
- [ ] Ensured that all commits in this PR are signed
- [ ] Ensured appropriate code coverage is maintained by unit tests

[ATB-1126]: https://govukverify.atlassian.net/browse/ATB-1126?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/manual-publish-alarm-to-dev.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          ref: ${{ inputs.gitRef }}
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@e1e17a757e536f70e52b5a12b2e8d1d1c60e04ef # pin@v1-node16
+        with:
+          role-to-assume: ${{ secrets.DEV_ALARM_GH_ACTIONS_ROLE_ARN }}
+          aws-region: eu-west-2
+
+      - name: "Deploy SAM app"
+        uses: alphagov/di-devplatform-upload-action@1db904c002e0f98dbcb5ab453ceb1b71de5f7dd3 # pin@v2
```
Comment (TaherKapasi) [inline .github/workflows/manual-publish-alarm-to-dev.yaml:50]: could we try using the govuk-one-login actions. 

https://github.com/govuk-one-login/devplatform-upload-action
https://github.com/govuk-one-login/devplatform-upload-action-ecr

eg `govuk-one-login/devplatform-upload-action-ecr` note they have dropped the `di-`

Comment (TaherKapasi) [reply .github/workflows/manual-publish-alarm-to-dev.yaml:50]: will also need to get the commit sha

File: .github/workflows/merge-alarm-to-main.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          ref: ${{ inputs.gitRef }}
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@e1e17a757e536f70e52b5a12b2e8d1d1c60e04ef # pin@v1-node16
+        with:
+          role-to-assume: ${{ secrets.ALARM_GH_ACTIONS_ROLE_ARN }}
+          aws-region: eu-west-2
+
+      - name: "Deploy SAM app"
+        uses: alphagov/di-devplatform-upload-action@1db904c002e0f98dbcb5ab453ceb1b71de5f7dd3 # pin@v2
```
Comment (TaherKapasi) [inline .github/workflows/merge-alarm-to-main.yaml:45]: same as above

**Review by TaherKapasi — CHANGES_REQUESTED** [review]

File: stack-orchestration/development_pipelines.sh [hunk 3]
```diff
@@ -7,5 +7,7 @@ AWS_ACCOUNT=di-id-reuse-core-dev-admin
 
 export AWS_ACCOUNT=$AWS_ACCOUNT
 
-./provisioner.sh $AWS_ACCOUNT ais-core-pipeline sam-deploy-pipeline v2.37.6 || exit 1
-./provisioner.sh $AWS_ACCOUNT ais-main-pipeline sam-deploy-pipeline v2.37.7 || exit 1
+#./provisioner.sh $AWS_ACCOUNT ais-core-pipeline sam-deploy-pipeline v2.37.6 || exit 1
```
Comment (domantas-gds) [inline stack-orchestration/development_pipelines.sh:10]: And these :)

**Review by domantas-gds — APPROVED** [review]

File: .github/workflows/manual-publish-alarm-to-dev.yaml (hunk 1 above)
Comment (domantas-gds) [reply .github/workflows/manual-publish-alarm-to-dev.yaml:50]: Shall we do this across VC Storage and AIS workflows in scope of a separate ticket?

Comment (domantas-gds) [reply .github/workflows/manual-publish-alarm-to-dev.yaml:50]: @TaherKapasi

**Review by TaherKapasi — APPROVED** [review]

**Comment by TaherKapasi** [comment]
Hi @MinaMahdi - Main SAM tempate has a warning -  W8001 Condition IsInternal not used

**Comment by MinaMahdi** [comment]
> Hi @MinaMahdi - Main SAM tempate has a warning - W8001 Condition IsInternal not used

Hey @TaherKapasi oddly enough the condition IsInternal is not used in my branch at the moment nor is it in our main branch from what I can see

**Pull request merged**